### PR TITLE
BB-4116: Add management command for hyperpay report

### PR DESCRIPTION
**Pull request by arjunsinghy96**
Adds a management command to report hyperpay manual review required ( see in [result codes](https://hyperpay.docs.oppwa.com/reference/resultCodes#successful)] type responses.

**Related Tickets**: [BB-4116](https://tasks.opencraft.com/browse/BB-4116)

### Testing Instructions:

1. Checkout to this branch
2. Get `ecommerce` running in `juniper` devstack (Preferable NELP devstack)
3. Attach to ecommerce using `make ecommerce-shell`

#### Unittest

1. Run unittests and verify that they pass (Use `./manage.py test extensions.payment.management.tests.test_hyperpay_report`)

#### Manual Test
1. Open shell using `./manage.py shell`
2. Create PaymentProcessorResponse objects by running the following code.
```
import datetime
from django.utils.timezone import now
from ecommerce.extensions.payment.models import PaymentProcessorResponse

PaymentProcessorResponse.objects.create(processor_name="hyperpay", response={"result": {"code": "000.400.000"}})
PaymentProcessorResponse.objects.create(processor_name="hyperpay_mada", response={"result": {"code": "000.400.000"}})
p1 = PaymentProcessorResponse.objects.create(processor_name="hyperpay", response={"result": {"code": "000.400.000"}})
p1.created = now() - datetime.timedelta(hours=9)
p1.save()
p2 = PaymentProcessorResponse.objects.create(processor_name="hyperpay_mada", response={"result": {"code": "000.400.000"}})
p2.created = now() - datetime.timedelta(hours=9)
p2.save()
```
3. exit shell
4. Run `./manage.py hyperpay_report --email email1@example.com email2@example.com`
5. Verify that email recipients contain email1@example.com email2@example.com
6. Verfiy that the email html has 2 table rows for bad responses
7.  Run `./manage.py hyperpay_report --duration 10 --email email1@example.com email2@example.com`
8. Verify that email recipients contain email1@example.com email2@example.com
6. Verfiy that the email html has 4 table rows for bad responses

Note: The email content will have lists with empty values as there are no Orders associated with them

### CODE REVIEW

File: ecommerce/extensions/payment/management/tests/test_hyperpay_report.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        )
+        with patch(
+            'ecommerce.extensions.payment.management.commands.hyperpay_report.Command._render_email_and_send'
+            ) as mock_method:
+            # We should only get 2 records for 5 hour timedelta
+            call_command('hyperpay_report', emails=["report1@example.com"])
+            mock_method.assert_called_with(*arguments_1)
+            # We should get 3 records now for 10 hours timedelta
+            call_command('hyperpay_report', emails=["report1@example.com"], duration=10)
+            mock_method.assert_called_with(*arguments_2)
```
Comment (0x29a) [inline ecommerce/extensions/payment/management/tests/test_hyperpay_report.py:113]: Nice tests. :+1:

File: ecommerce/extensions/payment/management/commands/hyperpay_report.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+from django.utils.timezone import now
+from oscar.core.loading import get_class, get_classes
+
+
+logger = logging.getLogger(__name__)
+PaymentProcessorResponse = get_class("payment.models", "PaymentProcessorResponse")
+HyperPay, HyperPayMada = get_classes("payment.processors.hyperpay", ["HyperPay", "HyperPayMada"])
+
+REPORT_DUARATION = 5
+SUCCESS_MANUAL_REVIEW_CODES_REGEX = re.compile(r'^(000\.400\.0[^3]|000\.400\.[0-1]{2}0)')
```
Comment (0x29a) [inline ecommerce/extensions/payment/management/commands/hyperpay_report.py:22]: nit: please add link to the docs with these codes listed.

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested this command as per PR description and got reports with different number of rows. Also, I ran `extensions.payment.management.tests.test_hyperpay_report` tests and ensured they're passing.
- [x] I read through the code.
- [x] Includes documentation.

File: ecommerce/extensions/payment/management/tests/test_hyperpay_report.py (hunk 1 above)
Comment (arjunsinghy96) [reply ecommerce/extensions/payment/management/tests/test_hyperpay_report.py:113]: Thanks Demid!

**Comment by giovannicimolin** [comment]
@arjunsinghy96 @0x29a @lgp171188 Aren't we using the pluggable payment processor for HyperPay? If so, I don't think these changes will have any effect.
If we're not using the pluggable approach, we should start migrating to it before the next named release.

**Comment by lgp171188** [comment]
@giovannicimolin, the codebase was converted to a pluggable app **after** this PR was merged. So, it should already have these changes.

**Comment by giovannicimolin** [comment]
@lgp171188 Got it, thanks for the context!

**Pull request merged**